types/defaultTo.d.ts

```diff
@@ -1,2 +1,4 @@
-export function defaultTo<T, U>(a: T, b: U | null | undefined): T | U;
-export function defaultTo<T>(a: T): <U>(b: U | null | undefined) => T | U;
+import { DefaultTo } from './util/tools';
+
+export function defaultTo<Fallback, Value>(a: Fallback, b: Value): DefaultTo<Fallback, Value>;
+export function defaultTo<Fallback>(a: Fallback): <Value>(b: Value) => DefaultTo<Fallback, Value>;
```

Comment:
We need 2 more things for this type

`Placeholder` variant
```typescript
export function defaultTo<Value>(__: Placeholder, b: Value): <Fallback>(a: Fallback) => DefaultTo<Fallback, Value>;
```
and to reorder them
```typescript
export function defaultTo<Fallback>(a: Fallback): <Value>(b: Value) => DefaultTo<Fallback, Value>;
export function defaultTo<Value>(__: Placeholder, b: Value): <Fallback>(a: Fallback) => DefaultTo<Fallback, Value>;
export function defaultTo<Fallback, Value>(a: Fallback, b: Value): DefaultTo<Fallback, Value>;
```

The re-ordering is because typescript always takes the _last_ overload when passing a function as an argument to another function (their docs explain it in the [conditional types](https://www.typescriptlang.org/docs/handbook/2/conditional-types.html) section actually)

So if we were to do `R.flip(R.defaultTo)` and have the `Placeholder` variant last, it would do that one, which is not desirable.

I'm starting to add tests for both `__` as well as with `pipe`, `compose`, and `flip` to make sure that behavior is expected. I you wouldn't mind taking care of this as well. Thank you!